src/core/expr.rs

```diff
@@ -39,6 +41,62 @@ impl From<Op> for utils::SExp {
     }
 }
 
+/// A custom function definition
+pub struct FnAssume
```

Comment:
More documentation is needed here. Does this represent the axioms for a particular function?